### ISSUE TITLE
Questions about Refinement Laplacian Regularization

### ISSUE THREAD

**Issue by violag** [wyysf-98/CraftsMan#8]
Thank you for the amazing work!

I was wondering how you're computing the relative Laplacian smoothing term. In the Palfinger paper, the number of vertices can change during the remesh steps. As a result wouldn't x and x_init be different sizes? Or are you keeping the number of vertices fixed?

<img width="875" alt="image" src="https://github.com/wyysf-98/CraftsMan/assets/11363939/50e73169-92ed-42d7-8df5-10cb694b525f">

**Comment by jtawade** [comment]
Just to clarify, x is the tensor of vertices that gets remeshed and has it's positions updated, while x_init is only remeshed during each training step. Is this correct?

Also in the original Continuous Remeshing paper you referenced, they update the gradients of the vertices using the Laplacian smoothing as follows:

https://github.com/Profactor/continuous-remeshing/blob/2e556421a6096701519c4f106d4af1f0588e19dd/core/opt.py#L126-L149

Wouldn't the gradient of equation 6 in your paper also have the same gradient as the gradient in the above linked code?

**Comment by violag** [comment]
Just to clarify, x is the tensor of vertices that gets remeshed and has it's positions updated, while x_init is only remeshed during each training step. Is this correct?

Also in the original Continuous Remeshing paper you referenced, they update the gradients of the vertices using the Laplacian smoothing as follows:

https://github.com/Profactor/continuous-remeshing/blob/2e556421a6096701519c4f106d4af1f0588e19dd/core/opt.py#L126-L149

Wouldn't the gradient of equation 6 in your paper also have the same gradient as the gradient in the above linked code?

**Comment by wyysf-98** [comment]
> https://github.com/Profactor/continuous-remeshing/blob/2e556421a6096701519c4f106d4af1f0588e19dd/core/opt.py#L126-L149

Hi, the gradient in the provided link enforces the smoothness between a vertex and its neighbors, whereas our smoothness term regularizes a vertex to be close to its initial position, so I think there are different.

**Comment by wyysf-98** [comment]
> > To solve this, we add another copy of mesh vertices and only do remesh operate and do **not** update its position.
> 
> Is x_init from the equation equivalent to the copy of mesh vertices where remeshing is done, but the positions aren't updated?

Yes, it is right

**Comment by liyukun-mathmagic** [comment]
but remesh opt will change vertices' position,how can it be  the copy of mesh vertices where remeshing is done, but the positions aren't updated?
